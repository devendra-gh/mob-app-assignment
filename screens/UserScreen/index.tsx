
import React from "react";
import { View, StyleSheet, Text, Image, SafeAreaView } from "react-native";
import { useTranslation } from "react-i18next";

import { ROUTES, setAuthData } from "../../utils/constant";
import CustomButton from "../../components/CustomButton";

const UserScreen = ({ route, navigation }: any) => {
  const { t } = useTranslation();
  const auth = route?.params?.auth || "";

  const handleLogout = async () => {
    await setAuthData(null);
    navigation.navigate(ROUTES.SIGNUP_ROUTE);
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.form}>
        <Image
          source={{ uri: "https://source.unsplash.com/random?wallpapers" }}
          style={{
            width: "100%",
            height: 250,
            alignSelf: "center",
            marginBottom: 15,
          }}
        />

        <View>
          <View style={styles.row}>
            <Text style={styles.head}>{t("name")}</Text>
            <Text style={styles.label}>{`${auth?.firstname || ""} ${auth?.lastname || ""}`}</Text>
          </View>
          <View style={styles.row}>
            <Text style={styles.head}>{t("email")}</Text>
            <Text style={styles.label}>{auth?.email || ""}</Text>
          </View>
        </View>

        <View style={styles.button}>
          <CustomButton title={t("logout")} onPress={handleLogout} />
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    paddingHorizontal: 20,
    backgroundColor: "#f5f5f5",
  },
  form: {
    backgroundColor: "#ffffff",
    padding: 20,
    borderRadius: 10,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  row: {
    justifyContent: "space-between",
    marginBottom: 10
  },
  head: {
    fontSize: 16,
    fontWeight: "bold",
  },
  label: {
    fontSize: 16,
  },
  button: {
    marginTop: 15
  },
});

export default UserScreen;
