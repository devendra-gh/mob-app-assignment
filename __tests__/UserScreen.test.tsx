import { render, act } from '@testing-library/react-native';
import UserScreen from "../screens/UserScreen";

describe('UserScreen', () => {
  it('should match snapshot', async () => {
    const result = render(
      <UserScreen />
    );
    await act(async () => { expect(result).toMatchSnapshot(); })
  });
});

