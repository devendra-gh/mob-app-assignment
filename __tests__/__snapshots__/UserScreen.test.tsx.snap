// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserScreen should match snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 10,
        "elevation": 5,
        "padding": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://source.unsplash.com/random?wallpapers",
        }
      }
      style={
        {
          "alignSelf": "center",
          "height": 250,
          "marginBottom": 15,
          "width": "100%",
        }
      }
    />
    <View>
      <View
        style={
          {
            "justifyContent": "space-between",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        />
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        >
           
        </Text>
      </View>
      <View
        style={
          {
            "justifyContent": "space-between",
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        />
        <Text
          style={
            {
              "fontSize": 16,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "marginTop": 15,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#0095fb",
            "borderRadius": 4,
            "marginTop": 10,
            "opacity": 1,
            "padding": 14,
          }
        }
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "700",
              "textAlign": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
